huggle/welcomeinfo.cpp

```diff
@@ -75,7 +102,8 @@ void Huggle::WelcomeInfo::on_label_2_linkActivated(const QString &link)
 void Huggle::WelcomeInfo::on_cb_Language_currentIndexChanged(const QString &arg1)
 {
     if (this->loading)  return;
-    QString lang = "en";
+    QLocale d;

```

Comment:
What is this code for? Maybe explanation in comment would be nice. I suppose you are trying to find the current OS locale and use it instead of hardcoded default. That is a good idea, but you should probably also check if such locale is present in Huggle's localization db. I have no idea what would happen if "lang" was a key that isn't present anywhere in LocalizationData.

Reply:
Also, I noticed that this code is really just used as a fallback in case that text in combobox doesn't really exist in LocalizationData. It's more of a failsafe in case that something really wrong happens somewhere. It's not default, so even if you replace lang = "en" with code you used, it wouldn't change the default selection which is defined in:

```
Localizations::HuggleLocalizations->PreferredLanguage
```